### PR TITLE
Feat: Added News page

### PR DESCRIPTION
**Pull request by vpranavkrishna**

### CODE REVIEW

File: lib/blocs/companies/companies_bloc.dart [hunk 1]
```diff
@@ -18,11 +18,9 @@ class CompaniesBloc extends Bloc<CompaniesEvent, CompaniesState> {
       try {
         final stockpricesStream =
             streamClient.getStockPricesUpdates(event.subscriptionId);
-        // ignore: avoid_print
-        print(event.subscriptionId);
+        //  print(event.subscriptionId);
         await for (final stockPrices in stockpricesStream) {
-          // ignore: avoid_print
-          print(stockPrices);
+          //  print(stockPrices);
```
Comment (dhruv1294) [inline lib/blocs/companies/companies_bloc.dart:23]: remove commented print statements

File: lib/blocs/market_events/market_event_event.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+class GetMarketEventFeed extends MarketEvent_events {
+  final SubscriptionId subscriptionId;
+  const GetMarketEventFeed(this.subscriptionId);
+  @override
+  List<Object> get props => [subscriptionId];
+}
+
+class GetMoreMarketEvent extends MarketEvent_events {
+  final int lasteventid;
```
Comment (dhruv1294) [inline lib/blocs/market_events/market_event_event.dart:22]: pls to follow camelcase

**Review by dhruv1294 — CHANGES_REQUESTED** [review]

**Review by Sudhindra3000 — APPROVED** [review]

**Review by dhruv1294 — APPROVED** [review]

File: lib/navigation/route_generator.dart [hunk 3]
```diff
@@ -76,6 +79,14 @@ class RouteGenerator {
         }
         throw Exception('Invalid phone args');
 
+      case '/news_page':
```
Comment (Sudhindra3000) [inline lib/navigation/route_generator.dart:82]:
```suggestion
      case '/news':
```

File: lib/blocs/market_events/market_event_event.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
+part of 'market_event_bloc.dart';
+
+abstract class MarketEventEvents extends Equatable {
```
Comment (Sudhindra3000) [inline lib/blocs/market_events/market_event_event.dart:3]: Rename this to `NewsEvent`
And rename the corresponding bloc and state classes to `NewsBloc` and `NewsState` as well

**Review by Sudhindra3000 — CHANGES_REQUESTED** [review]

File: lib/blocs/market_events/market_event_state.dart [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+class SubscriptionToMarketEventSuccess extends NewsState {
+  final MarketEventUpdate news;
+  const SubscriptionToMarketEventSuccess(this.news);
+
+  @override
+  List<Object> get props => [news];
+}
+
+class SubscriptionToMarketEventFailed extends NewsState {
```
Comment (Sudhindra3000) [inline lib/blocs/market_events/market_event_state.dart:33]: Rename these classes similarly

**Review by Sudhindra3000 — CHANGES_REQUESTED** [review]

File: lib/blocs/market_events/market_event_state.dart (hunk 5 above)
Comment (Sudhindra3000) [reply lib/blocs/market_events/market_event_state.dart:33]: And the names of the files and folders these bloc classes are in

**Comment by dhruv1294** [comment]
Also build is failing do fix it

**Review by Sudhindra3000 — APPROVED** [review]

**Review by dhruv1294 — APPROVED** [review]

**Pull request merged**